### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by k2jac9**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Added code to fit a KNN model using the best value for n_neighbors (determined through grid search) and then evaluated its performance on the test set using accuracy_score.

## What did you learn from the changes you have made?
The art and science of the trait. The KNN model was successfully trained with the optimal number of neighbors, and performance was evaluated on the test set, providing a metric to assess model effectiveness.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of? 
To use pipeline to optimize experimentation.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Implementing different distance metrics (Euclidean, Manhattan, Minkowski) alongside tuning n_neighbors, improve fitting and evaluating the model with the best n_neighbors. This was resolved by properly initializing the stratified_kfold before using it in GridSearchCV.

## How were these changes tested?
The changes were tested by running the fitting and evaluation code, ensuring that the accuracy score was correctly calculated for the test set based on the best hyperparameters from the grid search.

## Checklist
- [X] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by Dmytro-Bonislavskyi — APPROVED** [review]
Well done!
You just forgot one more way to count columns:
```

num_columns = wine_df.columns.nunique()
print(f"The Wine dataset contains {num_columns} variables.")
```

Just kidding! 😊

**Comment by k2jac9** [comment]
It has been added successfully! 😂
